### PR TITLE
[PD-7809] map markers

### PR DESCRIPTION
**Pull request by nate998877**
## What does this PR do?
Merges the 4 map components that previously existed into one. With the goal of creating one place to add bug fixes & extend functionality. The existing military & job maps have diverged from each other regarding features & functionality. 

## Notes to Reviewer
Testing/ examples available at https://github.com/DirectEmployers/microsite-template/tree/map-example

Please ensure `NUXT_ENV_GM_API_KEY` is set inside .env.development in the template. Otherwise you will receive 500 errors

### CODE REVIEW

File: src/components/Search/Maps/AppMap.vue [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+                return psr(coords.split(","))
+            } else if (this.markerConfig.coordParser) {
+                return this.markerConfig.coordParser(coords)
+            }
+
+            return false
+        },
+        assignIcons(marker){
+            // marker.icon = `https://d12wqovxet6953.cloudfront.net/jobmap/images/${marker.company_exact.toLowerCase()}.png`
+            marker.icon = "https://frontdoor.jobs/favicon.ico"
```
Comment (surgiie) [inline src/components/Search/Maps/AppMap.vue:317]: This shouldnt be hard coded to a specific company.

**Review by surgiie — CHANGES_REQUESTED** [review]

**Review by surgiie — CHANGES_REQUESTED** [review]
Minor code changes noted. Will re-review when PR/example is ready.

File: src/components/Search/Maps/AppMilitaryMap.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                this.done(markers)
+            } catch (error) {
+                this.failed(error)
+            }
+        },
+    },
+}
+</script>
+
+{
```
Comment (surgiie) [inline src/components/Search/Maps/AppMilitaryMap.vue:59]: Dont think you meant to leave this here?

**Review by surgiie — CHANGES_REQUESTED** [review]

**Review by surgiie — CHANGES_REQUESTED** [review]
* Clicking on a job map marker in your template example doesnt navigate to search page
* Im a bit hesistant to merge because of the use of "Extends" im under the impression this is not available or discouraged in vue 3.
* Your military example is also broken so unable to test that map:

image

* I would also prefer that @zarifrahat plays around with these examples, particularly the military one once its working since he is using it on sites currently.

**Comment by nate998877** [comment]
> * Clicking on a job map marker in your template example doesnt navigate to search page
> * Im a bit hesistant to merge because of the use of "Extends" im under the impression this is not available or discouraged in vue 3.
> * Your military example is also broken so unable to test that map:
> * I would also prefer that @zarifrahat plays around with these examples, particularly the military one once its working since he is using it on sites currently.

All of the maps work, it appears you maybe were missing an API key, I updated the comments to reflect that. I thought we all had API keys already set.

The job map has been updated to fix the navigation issues.

Zarif defiantly needs to play with these & I have added him as a reviewer.

**Review by zarifrahat — APPROVED** [review]

**Review by surgiie — APPROVED** [review]

**Review by jeremytubbs — APPROVED** [review]

**Pull request merged**